front-end/src/components/ClientRegisterForm.tsx

```diff
@@ -0,0 +1,79 @@
+"use client";
+
+import React from "react";
+import { useFormik } from "formik";
+import * as Yup from "yup";
+import { TextField, Button } from "@mui/material";
+
+const validationSchema = Yup.object({
+  email: Yup.string().email("Invalid email address").required("Required"),
+  password: Yup.string().required("Required"),
+});
+
+const ClientRegisterForm = () => {
+  const formik = useFormik({
+    initialValues: {
+      email: "",
+      password: "",
+    },
+    validationSchema,
+    onSubmit: async (values) => {
+      try {
+        const response = await fetch("http://localhost:5000/users", {
```

Comment:
We will create a list of function (or generate from open api) for the call then use react query.